### PR TITLE
tests, zone manager: Adding unit tests for zone manager component

### PR DESCRIPTION
**Pull request by dteplits**
Unit tests added for:
- ZoneManager
- ZoneFileCache
- ZoneFile

Signed-off-by: Diana Teplits <dteplits@redhat.com>

```release-note
None
```

### CODE REVIEW

File: pkg/zonemgr/zone_file_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	BeforeEach(func() {
+		zoneFile = NewZoneFile(zoneFileName)
+		Expect(os.Mkdir("zones", 0777)).To(Succeed())
+	})
+	AfterEach(func() {
+		Expect(os.RemoveAll("zones")).To(Succeed())
+	})
+
+	testFileFunc := func(fileContent string) {
```
Comment (AlonaKaplan) [inline pkg/zonemgr/zone_file_test.go:27]: I would change `fileContent` to `expectedFileContent`

File: pkg/zonemgr/zone_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	var zoneMgr *ZoneManager
+
+	BeforeEach(func() {
+		os.Setenv("DOMAIN", customDomain)
+		os.Setenv("NAME_SERVER_IP", customNSIP)
+		zoneMgr = NewZoneManager()
+	})
+
+	AfterEach(func() {
```
Comment (AlonaKaplan) [inline pkg/zonemgr/zone_manager_test.go:27]: There is no need to add an empty `AfterEach`.

File: pkg/zonemgr/zone_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	})
+
+	Context("Initialization", func() {
+		It("should validate input data", func() {
+			Expect(zoneMgr.UpdateZone(k8stypes.NamespacedName{Name: "vm1"}, nil)).NotTo(Succeed())
```
Comment (AlonaKaplan) [inline pkg/zonemgr/zone_manager_test.go:32]: Please split into two tests. `should fail updating a VMI with no name` and `should fail updating a VMI with no namespace`.

File: pkg/zonemgr/zone_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+			Expect(zoneMgr.UpdateZone(k8stypes.NamespacedName{Namespace: "ns1"}, nil)).NotTo(Succeed())
+		})
+
+		It("should set custom data", func() {
+			Expect(zoneMgr.UpdateZone(k8stypes.NamespacedName{Namespace: "ns1", Name: "vm1"}, nil)).To(Succeed())
```
Comment (AlonaKaplan) [inline pkg/zonemgr/zone_manager_test.go:37]: `UpdateZone` is updating the `domain` and `nameServerIP`? I would expect `NewZoneManager` to do it.

File: pkg/zonemgr/zone_manager_test.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+			Expect(zoneMgr.zoneFileCache.nameServerIP).To(Equal(customNSIP))
+		})
+
+		It("should create zone file with correct name", func() {
+			Expect(zoneMgr.zoneFile.zoneFileFullName).To(Equal("/zones/db.vm." + customDomain))
+		})
+
+		When("zone file already exist", func() {
+			It("should update cached SOA serial value", func() {
+				//todo
```
Comment (AlonaKaplan) [inline pkg/zonemgr/zone_manager_test.go:48]: TODO can be filtered by the IDE.

File: pkg/zonemgr/zone_manager_test.go (hunk 4 above)
Comment (dteplits) [reply pkg/zonemgr/zone_manager_test.go:37]: NewZoneManager indeed updates domain and nameServerIP. I use UpdateZone just to check that values are fetched and populated correctly

File: pkg/zonemgr/zone_file_cache_test.go [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+			aRecord_nic1_vm2_ns1 = fmt.Sprintf(aRecordFmt, nic1Name, vmi2Name, vmi1Namespace, nic1IP)
+			aRecord_nic2_vm2_ns1 = fmt.Sprintf(aRecordFmt, nic2Name, vmi2Name, vmi1Namespace, nic2IP)
+			aRecord_nic3_vm2_ns1 = fmt.Sprintf(aRecordFmt, nic3Name, vmi2Name, vmi1Namespace, nic3IP)
+			aRecord_nic4_vm2_ns1 = fmt.Sprintf(aRecordFmt, nic4Name, vmi2Name, vmi1Namespace, nic4IP)
+
+			aRecord_nic3_vm1_ns2 = fmt.Sprintf(aRecordFmt, nic3Name, vmi1Name, vmi2Namespace, nic3IP)
+			aRecord_nic4_vm1_ns2 = fmt.Sprintf(aRecordFmt, nic4Name, vmi1Name, vmi2Namespace, nic4IP)
+		)
+
+		validateUpdateFunc := func(newVmiName, newVmiNamespace string, newInterfaces []v1.VirtualMachineInstanceNetworkInterface,
```
[the remark this answers is not in the document]
Comment (dteplits) [reply pkg/zonemgr/zone_file_cache_test.go]: done

File: pkg/zonemgr/zone_file_cache_test.go [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+				zoneFileCache = NewZoneFileCache(nameServerIP, domain)
+				zoneFileCache.init()
+			})
+
+			DescribeTable("Updating interfaces records", validateUpdateFunc,
+				Entry("when new vmi with interfaces list is added",
+					vmi1Name,
+					vmi1Namespace,
+					[]v1.VirtualMachineInstanceNetworkInterface{{IPs: []string{nic1IP}, Name: nic1Name}},
+					true,
```
[the remark this answers is not in the document]
Comment (dteplits) [reply pkg/zonemgr/zone_file_cache_test.go]: done

File: pkg/zonemgr/zone_file_cache_test.go [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+			Entry("header should contain default values", "", "", headerDefault),
+			Entry("header should contain custom values", nameServerIP, domain, headerCustom),
+		)
+	})
+
+	Describe("cached zone file records update", func() {
+		const (
+			vmi1Name      = "vmi1"
+			vmi2Name      = "vmi2"
+			vmi1Namespace = "ns1"
```
[the remark this answers is not in the document]
Comment (dteplits) [reply pkg/zonemgr/zone_file_cache_test.go]: done

File: pkg/zonemgr/zone_file_test.go (hunk 1 above)
Comment (dteplits) [reply pkg/zonemgr/zone_file_test.go]: done

File: pkg/zonemgr/zone_manager_test.go (hunk 2 above)
Comment (dteplits) [reply pkg/zonemgr/zone_manager_test.go]: done

File: pkg/zonemgr/zone_manager_test.go (hunk 3 above)
Comment (dteplits) [reply pkg/zonemgr/zone_manager_test.go]: done

File: pkg/zonemgr/zone_manager_test.go (hunk 5 above)
Comment (dteplits) [reply pkg/zonemgr/zone_manager_test.go]: done

**Pull request merged**